src/code/z_player_lib.c

```diff
@@ -502,8 +502,16 @@ int Player_InCsMode(PlayState* play) {
     return Player_InBlockingCsMode(play, this) || (this->unk_6AD == 4);
 }
 
-s32 func_8008E9C4(Player* this) {
-    return (this->stateFlags1 & PLAYER_STATE1_4);
+/**
+ * Checks if currently locked onto a hostile actor.
+ * `PLAYER_STATE1_HOSTILE_LOCK_ON` controls Player's "battle" response to hostile actors.
+ * This includes affecting how movement is handled, and enabling a "fighting" set of animations.
+ *
+ * Note that `Player_UpdateHostileLockOn` exists, which updates this flag and also returns the check.
+ * This function should be used if the flag should be checked, but not updated.
```

Comment:
This comment makes me think that `Player_UpdateHostileLockOn` is available for everyone to be used instead of it being an internal player function.
Not sure how to better word it tho

Reply:
I see, yeah I didn't consider that `Player_CheckHostileLockOn` is public-facing. This comment is written from the perspective of player internally.
It would make sense if player only used his internal function, and external things only used this function, but that is not the case

Reply:
`Note that internally within Player, `Player_UpdateHostileLockOn` exists, which updates this flag and also returns the check.`

Reply:
I pushed something similar to this